brave-instrumentation/aws-java-sdk-v2-core/src/main/java/brave/instrumentation/awsv2/TracingExecutionInterceptor.java

```diff
@@ -79,55 +77,49 @@
   }
 
   /**
-   * Before an individual http request happens, may be run multiple times if retries occur
+   * Before an individual http request is finalized. This is only called once per operation, meaning
+   * we can only have one span per operation.
    */
   @Override public SdkHttpRequest modifyHttpRequest(
       Context.ModifyHttpRequest context,
       ExecutionAttributes executionAttributes
   ) {
     TraceContext maybeDeferredRootSpan = executionAttributes.getAttribute(DEFERRED_ROOT_CONTEXT);
-    Span applicationSpan;
+    Span span;
     HttpClientRequest request = new HttpClientRequest(context.httpRequest());
     if (maybeDeferredRootSpan != null) {
       Boolean sampled = httpTracing.clientRequestSampler().trySample(request);
       if (sampled == null) {
         sampled = httpTracing.tracing().sampler().isSampled(maybeDeferredRootSpan.traceId());
       }
-      applicationSpan = tracer.toSpan(maybeDeferredRootSpan.toBuilder().sampled(sampled).build());
-      executionAttributes.putAttribute(APPLICATION_SPAN, applicationSpan.start());
+      span = tracer.toSpan(maybeDeferredRootSpan.toBuilder().sampled(sampled).build());
+      executionAttributes.putAttribute(APPLICATION_SPAN, span.start());
     } else {
-      applicationSpan = executionAttributes.getAttribute(APPLICATION_SPAN);
+      span = executionAttributes.getAttribute(APPLICATION_SPAN);
     }
 
     String serviceName = executionAttributes.getAttribute(SdkExecutionAttribute.SERVICE_NAME);
     String operation = getAwsOperationNameFromRequestClass(context.request());
-    applicationSpan.name("aws-sdk")
+    span.name("aws-sdk")
         .tag("aws.service_name", serviceName)
         .tag("aws.operation", operation);
 
-    Span span = tracer.newChild(applicationSpan.context());
     handler.handleSend(request, span);
```

Comment:
probably we can do the normal http client stuff here instead of provisioning a span up front then passing it here.

ex `span = handler.handleSend(request);`